src/service/url-replacements-impl.js

```diff
@@ -445,94 +459,166 @@ export class UrlReplacements {
    * The data for the timing events is retrieved from performance.timing API.
    * If start and end events are both given, the result is the difference between the two.
    * If only start event is given, the result is the timing value at start event.
-   * @param {*} startEvent
-   * @param {*=} endEvent
-   * @return {!Promise<string|undefined>}
+   * @param {string} startEvent
+   * @param {string=} endEvent
+   * @return {!Promise<ResolverReturnDef>}
    * @private
    */
-  getTimingData_(startEvent, endEvent) {
-    const timingInfo = this.win_['performance']
-        && this.win_['performance']['timing'];
+  getTimingDataAsync_(startEvent, endEvent) {
+    let metric = this.getTimingDataSync_(startEvent, endEvent);
```

Comment:
> do not need to unnecessarily wait for load

We're returning a promise, though. Even if it's resolved sync (`Promise.resolve(metric)`), the caller can't get it's value until ~.1ms later. Adding another sync resolved Promise (`loadPromise(this.win_)`) only adds another ~.1ms delay.

This can definitely be in another PR, though.